data-prepper-plugins/s3-source/src/test/java/com/amazon/dataprepper/plugins/source/S3ObjectWorkerTest.java

```diff
@@ -172,6 +197,7 @@ void parseS3Object_throws_Exception_and_increments_counter_when_unable_to_get_S3
         assertThat(actualException, sameInstance(expectedException));
 
         verify(s3ObjectsFailedCounter).increment();
+        verifyNoInteractions(s3ObjectsSucceededCounter);
     }
```

Comment:
Does the timer still record a metric when parseS3Object throws an exception? This will probably impact the average read time but that should be apparent to users monitoring their metrics given we are reporting failures. 

Let's verify the interaction with the timer metric as well.